src/test/kotlin/uk/gov/justice/digital/hmpps/externalusersapi/resource/UserGroupControllerIntTest.kt

```diff
@@ -10,6 +10,111 @@ import uk.gov.justice.digital.hmpps.externalusersapi.integration.IntegrationTest
 
 class UserGroupControllerIntTest : IntegrationTestBase() {
 
+  @Nested
+  inner class AddGroupByUserId {
+
+    @Test
+    fun `access forbidden without admin role`() {
+      webTestClient.put().uri("/users/90F930E1-2195-4AFD-92CE-0EB5672DA44B/groups/SITE_1_GROUP_1")
+        .headers(setAuthorisation("bob", listOf()))
+        .exchange()
+        .expectStatus().isForbidden
+    }
+
+    @Test
+    fun `access forbidden without valid token`() {
+      webTestClient.put().uri("/users/90F930E1-2195-4AFD-92CE-0EB5672DA44B/groups/SITE_1_GROUP_1")
+        .exchange()
+        .expectStatus().isUnauthorized
+    }
+
+    @Test
+    fun `access forbidden with incorrect role`() {
+      webTestClient.put().uri("/users/90F930E1-2195-4AFD-92CE-0EB5672DA44B/groups/SITE_1_GROUP_1")
+        .headers(setAuthorisation("ITAG_USER_ADM", listOf("ROLE_MAINTAIN_EMAIL_DOMAINS")))
+        .exchange()
+        .expectStatus().isForbidden
+    }
+
+    @Test
+    fun `add group to a user success`() {
+      callGetGroups(userId = "7CA04ED7-8275-45B2-AFB4-4FF51432D1EC")
+        .jsonPath("[?(@.groupCode == 'SITE_1_GROUP_2')]")
+        .doesNotExist()
+
+      webTestClient
+        .put().uri("/users/7CA04ED7-8275-45B2-AFB4-4FF51432D1EC/groups/site_1_group_2")
+        .headers(setAuthorisation("ITAG_USER_ADM", listOf("ROLE_MAINTAIN_OAUTH_USERS")))
+        .exchange()
+        .expectStatus().isNoContent
+
+      callGetGroups(userId = "7CA04ED7-8275-45B2-AFB4-4FF51432D1EC")
+        .jsonPath("[?(@.groupCode == 'SITE_1_GROUP_2')]")
+        .isEqualTo(mapOf("groupCode" to "SITE_1_GROUP_2", "groupName" to "Site 1 - Group 2"))
+    }
+
+    @Test
+    fun `add group to a user as group manager`() {
```

Comment:
Should we test add group to a user that doesn't exist?

Reply:
A user that doesn't exist currently gives a 200 response. The service implementation is very similar to removing a group from a user, which I raised a ticket on. Reluctant to get into changing end point behaviour at this point. I'll add looking into this the ticket.

Reply:
I've expanded the scope of HAAR-944 to include this